tsp-output/@azure-tools/typespec-csharp/src/Generated/OpenAIClient.cs

```diff
@@ -13,6 +13,9 @@ namespace OpenAI
     /// <summary> The OpenAI service client. </summary>
     public partial class OpenAIClient
     {
+        protected ApiKeyCredential KeyCredential => _keyCredential;
+        protected Uri Endpoint => _endpoint;
```

Comment:
Some client fields become protected properties so that they're accessible to derived clients.  Consider this approach.